app/src/main/java/com/example/photofy/fragments/CameraFragment.java

```diff
@@ -0,0 +1,154 @@
+
+        executor = Executors.newSingleThreadExecutor();
+
+        previewView = view.findViewById(R.id.previewView);
+        fabImageCapture = view.findViewById(R.id.fabImageCapture);
+        cameraProviderFuture = ProcessCameraProvider.getInstance(getContext());
+
+        cameraProviderFuture.addListener(() -> {
+            try {
+                ProcessCameraProvider cameraProvider = cameraProviderFuture.get();
+                bindPreview(cameraProvider);
+            } catch (ExecutionException | InterruptedException e) {
+                e.printStackTrace();
+            }
+        }, ContextCompat.getMainExecutor(getContext()));
+
+        fabImageCapture.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                File file = getPhotoFile();
+                ImageCapture.OutputFileOptions outputFileOptions =
+                        new ImageCapture.OutputFileOptions.Builder(file).build();
+                imageCapture.takePicture(outputFileOptions, executor,
+                        new ImageCapture.OnImageSavedCallback() {
+                            @Override
+                            public void onImageSaved(ImageCapture.OutputFileResults outputFileResults) {
+                                Image picture = new Image();
+                                picture.setUser(ParseUser.getCurrentUser());
+                                picture.setImage(new ParseFile(file));
+                                picture.saveInBackground(new SaveCallback() {
+                                    @Override
+                                    public void done(ParseException e) {
+                                        // TODO: fix bundle and fragment, need to show image before picture completely finishes saving
+                                        CameraResultFragment cameraResultFragment = new CameraResultFragment();
+                                        Bundle bundle = new Bundle();
+                                        bundle.putParcelable("image", picture);
+                                        cameraResultFragment.setArguments(bundle);
+                                        ((MainActivity) getContext()).getSupportFragmentManager()
+                                                .beginTransaction()
+                                                .replace(R.id.flContainer, cameraResultFragment)
+                                                .addToBackStack(null)
+                                                .commit();
+                                    }
+                                });
+                                Log.i(TAG, "Image saved successfully");
+                            }
+                            @Override
+                            public void onError(ImageCaptureException error) {
+                                Log.e(TAG, "failed");
+                                error.printStackTrace();
```

Comment:
Replace this with Log statement i.e `Log.e(TAG, 'Error while capturing camera image', error)`.